docs/book/book.typ

```diff
@@ -0,0 +1,178 @@
+
+To achieve the goal of potentially replacing the existing `jInterface`, the project includes a split of the foundation layer into two parts: `core` and `otp`. This allows us to separate the core functionality of the application from the specific implementation details related to the communication protocol. It also makes it easier for developers to understand the structure of the project and make targeted modifications while reducing the potential impact on other parts of the system.
+
+
+= ZIO Integration
+
+ZIO is a modern, purely functional library for asynchronous and concurrent programming in Scala. It provides a powerful effect system that enables developers to write type-safe, composable, and resource-safe code without relying on traditional thread-based concurrency models. At its core, ZIO introduces fibers, which are lightweight, cooperative threads managed by the ZIO runtime. Fibers allow applications to achieve massive scalability and low latency while avoiding common pitfalls of concurrency such as deadlocks, race conditions, and resource leaks. Unlike traditional approaches that depend on locks or blocking operations, ZIO uses a lock-free concurrency model based on atomic operations and structured concurrency principles, ensuring predictable behavior and safe cancellation of tasks.
+
+ZIO also includes a rich ecosystem of libraries that extend its capabilities:
+
+- *ZIO Streams* for processing large or infinite data streams with backpressure and composable operators.
+- *ZIO Config* for type-safe configuration management from multiple sources.
+- *ZIO Logging* for structured, context-aware logging across asynchronous boundaries.
+- *ZIO STM* for software transactional memory, enabling safe and composable concurrent state updates without locks.
+- *ZIO Test* for deterministic testing of concurrent code and property-based testing.
+
+In our modernization project, ZIO forms the core of the Foundation Layer, providing structured concurrency, resource management, and effect handling. We leverage ZIO's fiber-based runtime to replace the legacy scheduling approach, ensuring that all operations run within well-defined scopes. This guarantees that resources like file handles, Lucene directories, and network sockets are properly acquired and released, even in the presence of failures. By using ZIO's ZManaged and finalizer constructs, we eliminate the risk of resource leaks and simplify error handling.
+
+The decision to adopt ZIO aligns with our goal of creating a modern, future-proof foundation without rewriting business logic. ZIO integrates seamlessly with Scala, allowing us to retain existing service classes while replacing the underlying concurrency and resource management layer. This approach provides immediate benefits—such as improved stability and observability—while laying the groundwork for future enhancements like reactive streaming, advanced scheduling.
```

Comment:
I would recommend some extra spaces added around the dashes for the "such as ... observability" part.